### PR TITLE
Execution commands to run scenario, specification and report issue 

### PR DESCRIPTION
**Pull request by BugDiver**
fix for following issues :

#24
#34
#37

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -44,7 +44,32 @@
...
+        "category": "Gauge"
+      },
+      {
+        "command": "gauge.execute.specification",
+        "title": "Run Specification",
+        "category": "Gauge"
+      },
+      {
+        "command": "gauge.execute.specification.all",
+        "title": "Run All Apecification",
```
Comment (zabil) [inline package.json]: Change Apecification -> Specification

File: package.json [hunk 2]
```diff
@@ -56,6 +81,13 @@
           "type": "boolean",
           "default": false,
           "description": "When true, communication between vscode and gauge server will be logged."
+        },
+        "gauge.execution.specDirs": {
+          "type": "array",
+          "default": [
+            "specs"
+          ],
+          "description": "List of specification dirs."
```
Comment (zabil) [inline package.json]: Change to 
List of specification directories.

File: src/execution/gaugeExecution.ts [hunk 3]
```diff
@@ -25,9 +30,79 @@ export function execute(spec: string, config: any) {
...
+export function runSpecification(all?: boolean): Thenable<any> {
+	if (all) {
+		let dirs = vscode.workspace.getConfiguration(GAUGE_EXECUTION_CONFIG).get<Array<string>>("specDirs");
+		return execute(dirs.join(" "), { inParallel: false });
+	} else {
+		let spec = vscode.window.activeTextEditor.document.fileName;
+		if (!extensions.includes(path.extname(spec))) {
+			vscode.window.showWarningMessage(`${spec} is not a valid specifcation.`);
+			return Promise.reject(new Error(`${spec} is not a valid specifcation.`));
+		} else {
```
Comment (nehashri) [inline src/execution/gaugeExecution.ts]: no need to have explicit else block.

File: src/execution/gaugeExecution.ts [hunk 4]
```diff
@@ -25,9 +30,79 @@ export function execute(spec: string, config: any) {
...
+		}
+	}
+};
+
+export function runScenario(languageClient: LanguageClient, atCusrsor: boolean): Thenable<any> {
+	let spec = vscode.window.activeTextEditor.document.fileName;
+	if (!extensions.includes(path.extname(spec))) {
+		vscode.window.showWarningMessage(`${spec} is not a valid specifcation.`);
+		return Promise.reject(new Error(`${spec} is not a valid specifcation.`));
+	} else {
```
Comment (nehashri) [inline src/execution/gaugeExecution.ts]: no need to have explicit else block.

File: src/extension.ts [hunk 5]
```diff
@@ -62,6 +62,28 @@ export function activate(context: ExtensionContext) {
...
+Gauge Extension Version: ${extVersion}
+${gaugeVersion.stdout.toString()}
+\`\`\``;
+
+    return opn(`https://github.com/getgauge/gauge-vscode/issues/new?body=${issueTempate}`).then(() => { }, (err) => {
+        vscode.window.showErrorMessage("Can't open issue URL. " + err);
+    });
+}
+
+function showReferences(languageClient: LanguageClient): any {
```
Comment (nehashri) [inline src/extension.ts]: showReferences function has been replaced with old code. Check master branch for correct code

**Pull request merged**